webhook/resources/volume/mutator.go

```diff
@@ -97,7 +97,24 @@ func (v *volumeMutator) Create(request *admission.Request, newObj runtime.Object
 	}
 
 	if string(volume.Spec.AccessMode) == "" {
-		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/accessMode", "value": "%s"}`, string(longhorn.AccessModeReadWriteOnce)))
+		accessModeFromBackup := longhorn.AccessModeReadWriteOnce
+		if volume.Spec.FromBackup != "" {
+			bName, _, _, err := backupstore.DecodeBackupURL(volume.Spec.FromBackup)
+			if err != nil {
+				return nil, werror.NewInternalError(fmt.Sprintf("cannot get backup and volume name from backup URL %v: %v", volume.Spec.FromBackup, err))
```

Comment:
```suggestion
				err := errors.Wrapf(err, "failed to decode backup url %v", volume.Spec.FromBackup)
				return nil, werror.NewInvalidError(err.Error(), "")
```

Reply:
Thanks, I will follow the suggestion to correct it.